esp-hal/src/hmac.rs

```diff
@@ -96,6 +96,8 @@ impl<'d> Hmac<'d> {
     pub fn new(hmac: impl Peripheral<P = HMAC> + 'd) -> Self {
         crate::into_ref!(hmac);
 
+        PeripheralClockControl::reset(PeripheralEnable::Sha);
```

Comment:
why does the HMAC driver tinker around with the SHA peripheral? Now that we reset it, it will also clash with the SHA driver

Reply:
The enable was there since the initial HMAC driver. I quickly checked with TRM (C6) and it says:

> 1. Enable HMAC module:
(a) Set the peripheral clock bits for HMAC and SHA peripherals in register SYSTEM_PERIP_CLK_EN1_REG, and clear the corresponding peripheral reset bits in register SYSTEM_PERIP_RST_EN1_REG. For information on those registers, please see Chapter 5 System and Memory.

However, in IDF they are not touching the SHA: https://github.com/espressif/esp-idf/blob/master/components/hal/esp32c6/include/hal/hmac_ll.h#L40-L52

Reply:
I think enabling is ~~fine~~ not a problem, but definitely not resetting - and if we do it the driver should take both peripherals (but as you said - ESP-IDF doesn't do that so maybe we shouldn't)